test/framework/fileConversion.go

```diff
@@ -37,44 +40,86 @@ func FormatTestData(srcFile, tgtDir string, targetFormats ...string) (string, er
 	return srcFile, nil
 }
 
-func tarCmd(src, tgtDir string) (string, error) {
-	base := filepath.Base(src)
-	tgt := filepath.Join(tgtDir, base+image.ExtTar)
-	args := []string{"-cf", tgt, src}
+func toTar(src, tgtDir string) (string, error) {
+	tgtFile, tgtPath, err := createTargetFile(src, tgtDir, image.ExtTar)
+	defer tgtFile.Close()
 
-	if err := doCmdAndVerifyFile(tgt, "tar", args...); err != nil {
-		return "", err
+	w := tar.NewWriter(tgtFile)
+	defer w.Close()
+
+	srcFile, err := os.Open(src)
+	if err != nil {
+		return "", errors.Wrap(err, "Error opening file")
 	}
-	return tgt, nil
+	defer srcFile.Close()
+
+	srcFileInfo, err := srcFile.Stat()
+	if err != nil {
+		return "", errors.Wrap(err, "Error stating file")
+	}
+
+	hdr, err := tar.FileInfoHeader(srcFileInfo, "")
+	if err != nil {
+		return "", errors.Wrap(err, "Error generating tar file header")
+	}
+
+	err = w.WriteHeader(hdr)
+	if err != nil {
+		return "", errors.Wrap(err, "Error writing header")
+	}
+	defer w.Close()
+
+	_, err = io.Copy(w, srcFile)
+	if err != nil {
+		return "", errors.Wrap(err, "Error writing to file")
+	}
+	return tgtPath, nil
 }
 
-func gzCmd(src, tgtDir string) (string, error) {
-	src, err := copyIfNotPresent(src, tgtDir)
+func toGz(src, tgtDir string) (string, error) {
```

Comment:
do you see a way to leverage the similar code between toTar, toGz and toXz?

Reply:
Will look to see what i can generalize

Reply:
There's not a real clean way to generalize the functions without hurting readability.  The pattern that each is written in is identical by design, but the calls they make are specialized and dissimilar in their returns.